synapse/storage/search.py

```diff
@@ -45,27 +45,29 @@ class SearchStore(BackgroundUpdateStore):
 
     def __init__(self, db_conn, hs):
         super(SearchStore, self).__init__(db_conn, hs)
-        self.register_background_update_handler(
-            self.EVENT_SEARCH_UPDATE_NAME, self._background_reindex_search
-        )
-        self.register_background_update_handler(
-            self.EVENT_SEARCH_ORDER_UPDATE_NAME,
-            self._background_reindex_search_order
-        )
 
-        # we used to have a background update to turn the GIN index into a
-        # GIST one; we no longer do that (obviously) because we actually want
-        # a GIN index. However, it's possible that some people might still have
-        # the background update queued, so we register a handler to clear the
-        # background update.
-        self.register_noop_background_update(
-            self.EVENT_SEARCH_USE_GIST_POSTGRES_NAME,
-        )
+        if hs.config.enable_search:
```

Comment:
You could change this into a `if not hs.config.enable_search: return` to avoid a big diff here.

Reply:
yes, although I didn't really want to risk people adding things to the end and wondering why they aren't be called in the future. Happy to switch it around though.

Reply:
We'll be able to catch cases like that with tests, I hope :)

Reply:
done